src/main/java/com/iexec/core/task/TaskService.java

```diff
@@ -460,4 +462,17 @@ private void resultUploaded2Finalized2Completed(Task task) {
         updateTaskStatusAndSave(task, COMPLETED);
         applicationEventPublisher.publishEvent(new TaskCompletedEvent(task));
     }
+
+    public void initializeTaskAccessForNewReplicate(String chainTaskId) {
+        taskAccessForNewReplicateLock.putIfAbsent(chainTaskId, false);
+    }
+
+    public Boolean isTaskAccessedForNewReplicate(String chainTaskId) {
```

Comment:
I'd got with isTaskBeingAccessedForNewReplicate()

Reply:
Done, much cleaner thank you